### ISSUE TITLE
Access to installer settings

### ISSUE THREAD

**Issue by s-hager** [gluap/pyess#17]
Hello!
@gluap 
I saw that in the disclaimer you say that you only have customer-level access to your LG ESS inverter. 
I actually found on a website that the password protected installer settings in the LG EnerVu Plus 1.3.2 android app can be accessed by typing in the "Registration NO." that is written on the right side of the inverter on the same sticker as the _WLAN Password_. just below it. My _Registration NO._ for example is a 16 character sequence that starts like  _DE2011B..._. The last 6 characters are numbers.

For anyone who is reading this just be careful not to change anything if you do not know exactly what you are doing because changes made in this menu can have more significant consequences.

**Comment by PanicMan** [comment]
Is it possible to read the SOH value out from the installer Battery info an put it to the MQTT?

**Comment by PanicMan** [comment]
I understand it, and I would assume to make an switch in the settings to explicitly activate it. The reason I need it, is, that my system is nearly 4 years old, and if I look from time to time into the installer settings, it shows me an SOH of 88%. But if I measure it on my system and count the amount of energy it discharges from 100-0% it is about 6.1kWh of an (usable) 9.3kWh Battery what is about 66% and that would definitely affect the warranty. Thats why I want to track the SOH to see if the BMS is somehow recognizing the missing energy...

**Comment by gluap** [comment]
As it is currently winter in the northern hemisphere: if you have winter mode enabled that might explain the discrepancy: winter mode essentially always keeps the battery charged to 20%, so in winter, only 80% of full capacity are usable.(at least on my system that's what it seems to be doing with default settings).

Depending on how winter mode accounts for soh, with 88%soh and 20% winter reserve only 68% capacity might be available.

**Comment by PanicMan** [comment]
Yes, I know that mode, and I already removed the additional 15% which are reserved in the Winter Mode. I have that issue already since the last summer. What I also encounter, is, that the Charge/Discharge Energy of the Battery is going apart and the Battery is getting warmer... In January f.e. the Battery charged 121.85 kWh and discharged 95.37 kWh, so where are the missing 26.48 kWh?
In my opinion the battery will soon die and I hope not with fire, as this are LiPo's... :(

**Comment by gluap** [comment]
For me in winter the numbers are similar in terms of lost energy. For instance 32 kWh battery charge vs 23 kWh battery discharge in December 2022, but it looked better again in July 2023: 138 kWh charge vs 137 kWh discharge. Bad again in December 2023: 22kWh battery charge vs 14 kWh discharge or January 2024 59 kWh charge vs 49 kWh battery discharge. My SOH is 94, battery from 2019.

A possible explanation aside from a broken battery would be the inverters own power consumption might go uncounted by LGs power computation - In summer that self-consumption is largely directly from PV, in winter with very little sun it's mostly from battery. One thing that I can't comment on is the battery temperature, I've not monitored that in the past, and the battery getting hot definitely sounds concerning.

**Comment by s-hager** [comment]
My SOH is at 88,5%, battery was installed in April 2021, manufactured September 2020. I also have backup-mode enabled (meaning it does not discharge below 30%). Just as an additional data input.

**Comment by PanicMan** [comment]
I can only say what I can see, and that is, that my battery capacity is decreasing rapidly from last summer and the battery is getting warmer and my house is an wood house what scares me a little... I need to track something, because the app is lying with SOH and the LG Service is not a help at all... In my opinion the Warranty promise from LG is an complete lie... I just hope that it will not end in an disaster...

**Comment by SamA699** [comment]
My SOH now shows 91% within 3 1/2 year.
I hope that monitoring this over the time with my homeassistant would give me more arguments in discussion with the LG-Service, when the battery shows below 80% for a possible guarantee change.
So, for me also, access to the installer menu with getting the SOH over mqtt would help.

**Comment by gluap** [comment]
If anyone wants to have a go, here's a short talk about how i reverse engineered the lg ess api for the user mode: https://youtu.be/C8qPi5ViYLA?si=Sad9u6iriGE_52fI
